assets/aml-benchmark/components/src/utils/online_endpoint/online_endpoint_model.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Class for online endpoint model."""
+
+
+from typing import Optional
+from ..logging import get_logger
+
+
+logger = get_logger(__name__)
+
+
+class OnlineEndpointModel:
+    """Class for online endpoint model."""
+
+    def __init__(
+            self, model: str, model_version: Optional[str], model_type: str,
+            endpoint_url: Optional[str] = None
+    ):
+        """Init method."""
+        if model is not None and model.startswith('azureml:'):
+            self._model_name = model.split('/')[-3]
+            self._model_path = model
+            if model_version is None:
+                self._model_version = model.split('/')[-1]
+        else:
+            self._model_name = model
+            self._model_path = None
+            self._model_version = model_version
+        self._model_type = model_type
+        if model_type is None:
+            self._model_type = self._get_model_type_from_url(endpoint_url)
+
+    @property
+    def model_name(self) -> str:
+        """Get the model name."""
+        return self._model_name
+
+    @property
+    def model_version(self) -> str:
+        """Get the model version."""
+        return self._model_version
+
+    @property
+    def model_type(self) -> str:
+        """Get the model type."""
+        return self._model_type
+
+    @property
+    def model_path(self) -> str:
+        """Get the model path."""
+        if self._model_path is None:
+            self._model_path = 'azureml://registries/azureml-meta/models/{}/versions/{}'.format(
```

Comment:
Are we assuming here that the model is a meta model and will be from azureml-meta registry?

Reply:
for the current release. Yes.
But as this parameter is not used in the customer facing pipeline, if not meta model, current workaround is to provide full path.